content/articles/creating-a-shopping-cart-api/index.md

```diff
@@ -0,0 +1,592 @@
+        cart.addProductToCartByCustomer(1L, product2);
+        cart.addProductToCartByCustomer(1L, product);
+        assertEquals(3,cart.totalNumberOfItems());
+
+        assertEquals(new BigDecimal( 3100), cart.calculatePrice());
+        customerDao=null;
+    }
+    @Test
+    void addProductToCartByAdmin(){
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product);
+        assertEquals(3,cart.totalNumberOfItems());
+        cart=null;
+    }
+    @Test
+    void removeProductsFromCartByAdmin(){
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product);
+        System.out.println(cart);
+        assertEquals(3,cart.totalNumberOfItems());
+        cart.removeProductsFromCartByAdmin(1L,product);
+        System.out.println(cart);
+        assertEquals(2,cart.totalNumberOfItems());
+
+    }
+    @Test
+    void checkAllTheProductsInTheCartByCustomer(){
+        Product product = new Product("1", "Sandine", new BigDecimal(1000));
+        Product product2 = new Product("2", "SandineBread", new BigDecimal(1100));
+        Product product3 = new Product("3", "SandineBread", new BigDecimal(1100));
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        cart.addProductToCartByAdmin(1L, product);
+        cart.addProductToCartByAdmin(1L, product2);
+        cart.addProductToCartByAdmin(1L, product3);
+        System.out.println(  cart.checkAllTheProductsInTheCartByCustomer(1L));
+        cart=null;
+    }
+    @Test
+    void checkAllRegisteredAdmin(){
+        Admin admin=new Admin(1L, "Kingsley","Nwafor");
+        dataService.registerAdmin(admin);
+        System.out.println(dataService.getAllAdmin());
+    }
+}
+```
+### Using Postman to verify the output on the web
+![shopping_api](/creating-a-shopping-cart-api/api.JPG)
+![shopping_api1](/creating-a-shopping-cart-api/api2.JPG)
+![shopping_api2](/creating-a-shopping-cart-api/payment.JPG)
+### Conclusion
+In the end, we've seen how to code to an interface, how to write a unit test, how to create an endpoint, and also how to use postman to view our endpoint. You may go on and start creating different APIs. Once again you can check [Github](https://github.com/kingsleynwafor54/shopping_cart_with_springboot)
```

Comment:
The last two sentences in this conclusion don't flow very well. For the second last sentence here it kind of falls flat flow-wise in that it feels like there is more to talk about (i.e. a more specific suggestion of how to use the knowledge from this guide). Also, in the last sentence, you don't specify what that Github link is for so it sounds weird. Please revise this.

Reply:
Thanks, I'll work on the corrections.